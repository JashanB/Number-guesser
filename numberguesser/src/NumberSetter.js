import React, { useState } from "react";


export default function NumberSetter (props) {
  return (
    <div className="set-your-number">
    <h3>Change the range: </h3>
    <form autoComplete="off" onSubmit={event => event.preventDefault()}>
      <input
        className="number-setter"
        type="number"
        placeholder="Guess a number!"
        onChange={event => {
          props.setRange(event.target.value);
        }
        }
        value={props.range}
        data-testid="number-setter-form"
      />
    </form>
    </div>
  )
};