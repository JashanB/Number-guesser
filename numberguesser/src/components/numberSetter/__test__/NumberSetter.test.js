import React from "react";
import ReactDOM from 'react-dom';
import NumberSetter from './../NumberSetter';
import { render, fireEvent, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

afterEach(cleanup);
it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<NumberSetter></NumberSetter>, div)
});

it ("renders correctly with passed down input", () => {
  const setter = render(<NumberSetter range={5}></NumberSetter>);
  const input = setter.getByTestId('number-setter-form');
  expect(input.value).toBe("5");
});

it("matches snapshot", () => {
  const tree = renderer.create(<NumberSetter range={5}></NumberSetter>).toJSON();
  expect(tree).toMatchSnapshot();
});


