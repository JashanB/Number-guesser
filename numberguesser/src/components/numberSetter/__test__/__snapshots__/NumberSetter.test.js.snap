// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="set-your-number"
>
  <h3>
    Change the range: 
  </h3>
  <form
    autoComplete="off"
    onSubmit={[Function]}
  >
    <input
      className="number-setter"
      data-testid="number-setter-form"
      onChange={[Function]}
      placeholder="Guess a number!"
      type="number"
      value={5}
    />
  </form>
</div>
`;
